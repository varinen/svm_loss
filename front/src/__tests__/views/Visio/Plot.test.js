import React from "react";
import Plot from "views/SVMLoss/Demo/Visio/Plot";
import {createShallow} from '@material-ui/core/test-utils';

describe('Plot renders correctly', () => {
    let component, props, shallow;
    beforeEach(() => {
        props = {
            plot: ''
        };
        shallow = createShallow({dive: true});
        component = shallow(<Plot {...props}/>);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

});


describe('Plot renders a loading component when the plot is empty', () => {
    let component, props, shallow;
    beforeEach(() => {
        props = {
            plot: ''
        };
        shallow = createShallow({dive: true});
        component = shallow(<Plot {...props}/>);
    });

    it('renders no plot', () => {
        expect(component.find('#plot').length).toBe(0)
    });

    it('renders a loading animation', () => {
        expect(component.find('#loading-plot').length).toBe(1)
    });

});


describe('Plot renders a plot image when the plot is not empty', () => {
    let component, props, shallow;
    beforeEach(() => {
        props = {
            plot: 'some plot'
        };
        shallow = createShallow({dive: true});
        component = shallow(<Plot {...props}/>);
    });

    it('renders no loader', () => {
        expect(component.find('#loading-plot').length).toBe(0)
    });

    it('renders a plot', () => {
        expect(component.find('#plot').length).toBe(1)
    });

});
