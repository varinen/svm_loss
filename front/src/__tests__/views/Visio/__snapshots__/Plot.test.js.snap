// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plot renders correctly renders correctly 1`] = `
<article>
  <WithStyles(ForwardRef(CircularProgress))
    id="loading-plot"
  />
</article>
`;
